perfkitbenchmarker/vm_util.py

```diff
@@ -501,3 +504,51 @@ def GenerateSSHConfig(vms):
     ofp.write(template.render({'vms': vms}))
   logging.info('ssh to VMs in this benchmark by name with: '
                'ssh -F {0} <vm name>'.format(target_file))
+
+
+@contextlib.contextmanager
+def RunDStatIfConfigured(vms):
+  """Installs and runs dstat on 'vms' if FLAGS.dstat is set.
+
+  On exiting the context manager, the results are copied to the run temp dir.
+
+  Args:
+    vms: List of virtual machines.
+
+  Yields:
+    None
+  """
+  if not FLAGS.dstat:
+    yield
+    return
+
+  lock = threading.Lock()
+  pids = {}
+  file_names = {}
+
+  def StartDStat(vm):
+    vm.Install('dstat')
+    dstat_file = os.path.join(VM_TMP_DIR, '{0}-dstat.csv'.format(vm.name))
+    cmd = ('nohup dstat --epoch --all --mem --vm --unix -o {0} '
+           '> /dev/null 2>&1 & echo $!').format(dstat_file)
```

Comment:
Another suggestion. It'd be great if we could collect error metrics too (I'm not entirely sure if dstat already does). I'm looking for metrics like dropped packets, TCP re-transmissions and timeouts, storage IO errors, etc. 
This follows the USE method methodology, which monitors utilization, saturation, and error metrics for each main component in a system, and serves as checklist to quickly identify bottlenecks.

Here's more info: http://www.brendangregg.com/USEmethod/use-linux.html